### PR TITLE
Gas Turbine Engines

### PR DESCRIPTION
**Pull request by EvilSpaceMantis**
#### Summary

```SUMMARY: Content "Adds new large gas turbine engines for military vehicles"```
#### Purpose of change

Adds three gas turbine engines for large vehicles to the game, eliminating the need to string multiple smaller engines together to power armored vehicles.

#### Describe the solution

Specific Changes:

Changes the Light Tank from mods/Tank to use twin existing v12 engines instead of 4 v8s as a realism update.
Adds a new 1350 hp turbine engine which spawns in the Main Battle Tank from mods/Tank (and switches the fuel tanks to gasoline to match the engine).
Adds a new 1900 hp turbine engine which spawns in the Apache and UH-60 wrecks in vehicles/helicopters.json
Adds a new 6000 hp turbine engine which spawns in the Osprey wrecks in vehicles/helicopters.json

All turbine engines are less fuel efficient than equivalent gas engines and require higher mechanics skill than existing engines (7, 8, and 9 respectively).

#### Describe alternatives you've considered

Current power outputs for the engines are real world accurate, but may be a little too powerful for how the game interprets them. For example, the Main Battle Tank with the new engine has a top speed of 80 mph where we would expect a top speed of 55-60.

Turbine engines provide small to medium quantities of superalloy sheets when destroyed at the moment to represent the advanced materials needed for the turbine blades. I am not sure if this would have any negative balance impacts though as it would make salvaging superalloy from helicopter crashes pretty simple in the early-mid game.

#### Additional context

From a discussion on the subreddit found here: https://www.reddit.com/r/cataclysmdda/comments/bbhqll/new_players_come_here_weekly_questions_and/ekqapyb?utm_source=share&utm_medium=web2x

### CODE REVIEW

File: data/json/vehicleparts/combustion.json [hunk 1]
```diff
@@ -36,6 +36,15 @@
     "//": "30% energy efficiency",
     "description": "A combustion engine.  Burns gasoline fuel from a tank in the vehicle."
   },
+  {
+    "abstract": "gas_turbine_engine",
+    "copy-from": "engine_combustion",
+    "type": "vehicle_part",
+    "fuel_type": "gasoline",
+    "m2c": 60,
```
Comment (mlangsdorf) [inline data/json/vehicleparts/combustion.json]: It might be worthwhile to lower the m2c here to 45 or, further limiting the amount of total engine power than can be used for safe velocity.  BUt it's probably not necessary since the speed problems come from the MBT being underweight.

File: data/json/vehicleparts/combustion.json [hunk 2]
```diff
@@ -36,6 +36,15 @@
...
+    "//": "25% energy efficiency",
+    "description": "An advanced combustion engine.  Burns gasoline fuel from a tank in the vehicle. Better power-to-weight ratio than a traditional reciprocating engine, but consumes more fuel."
```
Comment (mlangsdorf) [inline data/json/vehicleparts/combustion.json]: Most turbines are multifuel.  The Abrams tank engine famously is, for instance.

File: data/json/vehicleparts/combustion.json [hunk 3]
```diff
@@ -36,6 +36,15 @@
     "//": "30% energy efficiency",
     "description": "A combustion engine.  Burns gasoline fuel from a tank in the vehicle."
   },
+  {
+    "abstract": "gas_turbine_engine",
+    "copy-from": "engine_combustion",
+    "type": "vehicle_part",
+    "fuel_type": "gasoline",
```
Comment (mlangsdorf) [inline data/json/vehicleparts/combustion.json]: jp8 (the default fuel for these things) is closer to diesel than gasoline, but has an energy density closer to gasoline.  I could go either way on the default fuel.

**Comment by thethunderhawk** [comment]
Maybe it should drop superalloy playing rather than sheets, so it can’t be used to make the superalloy MBR vest too early?

File: data/json/items/vehicle/engine.json [hunk 4]
```diff
@@ -257,5 +257,38 @@
     "weight": 8000,
     "volume": 2,
     "price": 3500
+  },
+  {
+    "id": "small_turbine_engine",
+    "copy-from": "engine_gasoline",
+    "type": "ENGINE",
+    "name": "1350 hp gas turbine engine",
+    "description": "A gas turbine engine, usually used for military vehicles. Known for its high rate of fuel consumption.",
```
Comment (Rail-Runner) [inline data/json/items/vehicle/engine.json]: Need to use double spaces between sentences in description.
```suggestion
    "description": "A gas turbine engine, usually used for military vehicles.  Known for its high rate of fuel consumption.",
```

Comment (EvilSpaceMantis) [reply data/json/items/vehicle/engine.json]: Spaces added, and I have switched them over to dropping fewer alloy plates rather than a larger number of alloy sheets.

File: data/json/vehicleparts/combustion.json (hunk 1 above)
Comment (EvilSpaceMantis) [reply data/json/vehicleparts/combustion.json]: I've dropped the light tank down to a single diesel v12.

Most elegant way I can see to fix the MBT weight issue would be to create a heavy_plating_military that doesn't perform a whole lot better but just weighs about 2x as much. I think I'll leave that out of this PR but stick it on a to-do list for making a second pass at armored vehicles in the future.

File: data/json/vehicleparts/combustion.json (hunk 2 above)
Comment (EvilSpaceMantis) [reply data/json/vehicleparts/combustion.json]: How multi-fuel do you want to go? I've changed the engines over to also accepting diesel and lamp oil, though left the spawning vehicle tanks filled with gas. Looks like it could run on methanol and ethanol as well but the fuel consumption would be be crazy.

File: data/json/vehicleparts/combustion.json (hunk 1 above)
Comment (mlangsdorf) [reply data/json/vehicleparts/combustion.json]: Good solution.

File: data/json/vehicleparts/combustion.json (hunk 2 above)
Comment (mlangsdorf) [reply data/json/vehicleparts/combustion.json]: Up to you.  Diesel engines run on methanol and ethanol, but each option adds another line to the engine control menu for each engine.

**Comment by ymber** [comment]
25% fuel efficiency feels low. From the data in this paper it looks like it should be around or over 50%.
https://i.stack.imgur.com/cnWLV.png
https://www.nap.edu/read/23490/chapter/6#38

**Comment by mlangsdorf** [comment]
From Table 3.6 of that paper, turboprops in the 1000 kW to 10000 kW range have a 25-35% thermal efficiency.  And that's turboprops used to spin propellers or act as part of jet engines; turbines driving a ground vehicle drive shaft have some different efficiency curves.  

Turbines in general have weird efficiency curves, because you want the turbine to spinning fairly fast for maximum efficiency.  CDDA doesn't calculate engine rotations yet, so it's impossible to say if an turbine is operating in the good part of its efficiency range.

The wikipedia article on the Abrams says it consumes 230L of JP8 (basically gasoline for CDDA's purposes) per hour traveling cross-country at 30 mph and 38L per hour while idling.  The proposed diesel engine consumes ~195L and 32L, respectively (14% less) so I feel pretty good with an effective turbine efficiency of 25% versus the diesel's effective efficiency around 40%.

File: data/json/items/vehicle/engine.json [hunk 5]
```diff
@@ -56,7 +56,7 @@
     "weight": 20000,
     "volume": 6,
     "price": 10000,
-    "displacement": 40,
+    "displacement": 60,
```
Comment (mlangsdorf) [inline data/json/items/vehicle/engine.json]: Hey, we just fixed these!  Don't change them.

File: data/json/vehicles/helicopters.json [hunk 6]
```diff
@@ -456,7 +459,7 @@
       { "x": -1, "y": 1, "chance": 7, "items": [ "mre_hotdog_box" ] },
       { "x": -1, "y": 1, "chance": 7, "items": [ "mre_chicken_box" ] },
       { "x": -1, "y": 1, "chance": 10, "items": [ "canteen" ] },
-      { "x": -1, "y": 1, "chance": 10, "items": [ "light_plus_battery_cell" ] },
+      { "x": -1, "y": 1, "chance": 10, "items": [ "battery" ] },
```
Comment (mlangsdorf) [inline data/json/vehicles/helicopters.json]: Revert this change.

File: data/json/items/vehicle/engine.json (hunk 5 above)
Comment (Rail-Runner) [reply data/json/items/vehicle/engine.json]: Might be accidental reverts, a leftover from resolving merge conflicts. Happened to me too.

Comment (EvilSpaceMantis) [reply data/json/items/vehicle/engine.json]: Yeah, must have been some weird merging. Fixed all the changed displacements and the battery overwrites to match the master.

**Comment by Mcrone617** [comment]
Hey @EvilSpaceMantis , i'm working with @DracoGriffin to try to increase the community involvement in the game, specifically on the subreddit. If you're active there you've probably seen the PR posts, asking for the community to help test out and leave feedback. I've checked out your PR, and it looks ideal for this, it's easy enough to test via debug, and the more feedback the better when it comes to merging PR's. I was thinking some sort of challenge to make some ridiculous debug vehicle, and try to see how fast you can go, and how balanced the new engines are. If you're interested, let me know either here, or i'm usually on the discord. All we need from you is to make a downloadable release build through GitHub ( I can help you out on this if you don't know how), and then I'll post it on the subreddit, and make a summary/challenge for it. (or you can post it, whatever you prefer) Let me know if you're interested!
